% !TEX TS-program = pdflatex
% !TEX encoding = UTF-8 Unicode

% This is a simple template for a LaTeX document using the "article" class.
% See "book", "report", "letter" for other types of document.

\documentclass[11pt]{article} % use larger type; default would be 10pt


\usepackage{ulem}
\newcommand\NoIndent[1]{%
  \par\vbox{\parbox[t]{\linewidth}{#1}}%
}


\usepackage[utf8]{inputenc} % set input encoding (not needed with XeLaTeX)

%%% Examples of Article customizations
% These packages are optional, depending whether you want the features they provide.
% See the LaTeX Companion or other references for full information.

%%% PAGE DIMENSIONS
\usepackage{geometry} % to change the page dimensions
\geometry{a4paper} % or letterpaper (US) or a5paper or....
% \geometry{margin=2in} % for example, change the margins to 2 inches all round
% \geometry{landscape} % set up the page for landscape
%   read geometry.pdf for detailed page layout information

\usepackage{graphicx} % support the \includegraphics command and options

% \usepackage[parfill]{parskip} % Activate to begin paragraphs with an empty line rather than an indent

%%% PACKAGES
\usepackage{booktabs} % for much better looking tables
\usepackage{array} % for better arrays (eg matrices) in maths
\usepackage{paralist} % very flexible & customisable lists (eg. enumerate/itemize, etc.)
\usepackage{verbatim} % adds environment for commenting out blocks of text & for better verbatim
\usepackage{subfig} % make it possible to include more than one captioned figure/table in a single float
% These packages are all incorporated in the memoir class to one degree or another...

%%% HEADERS & FOOTERS
\usepackage{fancyhdr} % This should be set AFTER setting up the page geometry
\pagestyle{fancy} % options: empty , plain , fancy
\renewcommand{\headrulewidth}{0pt} % customise the layout...
\lhead{}\chead{}\rhead{}
\lfoot{}\cfoot{\thepage}\rfoot{}

%%% SECTION TITLE APPEARANCE
\usepackage{sectsty}
\allsectionsfont{\sffamily\mdseries\upshape} % (See the fntguide.pdf for font help)
% (This matches ConTeXt defaults)

%%% ToC (table of contents) APPEARANCE
\usepackage[nottoc,notlof,notlot]{tocbibind} % Put the bibliography in the ToC
\usepackage[titles,subfigure]{tocloft} % Alter the style of the Table of Contents
\renewcommand{\cftsecfont}{\rmfamily\mdseries\upshape}
\renewcommand{\cftsecpagefont}{\rmfamily\mdseries\upshape} % No bold!

%%% END Article customizations


\usepackage{verbatim}
\usepackage{amsmath}






\title{Work Log for August}
\author{Logan Brown}
%\date{} % Activate to display a given date or no date (if empty),
         % otherwise the current date is printed 

\begin{document}
%\maketitle
%\tableofcontents
%\newpage


\newpage
\setcounter{section}{17}
\setcounter{subsection}{-1}
\setcounter{subsubsection}{0}

\section{August 18th}
\subsection{Goals for the day}
\textit{Goals from Last Time}
\begin{enumerate}
\item \textit{Analyze my.cubappr.r}
\begin{enumerate}
\item \textit{Add in print statements, then rerun the example}
\item \textit{my.drawBConditionalAll.??????}
\end{enumerate}

\item \textit{(Optional) Study R user manual more?}

~

\NoIndent{Additional Goals}
\item \textbf{Sign up and connect GitHub} (TOP PRIORITY)
\item LaTeX changes
\item \textbf{Read Gilchrist 2014 paper} (HIGH PRIORITY)
\item Look at REU Results (a1 is questionable?)
\item Workflow Tracker (like doxygen) for R?

\end{enumerate}

\subsection{Progress/Notes}

\subsubsection{Analyze my.cubappr.r}

\subsubsection{Study R user manual}

\subsubsection{Connect GitHub}

Connected. Username ozway. Forked cubfits. Added "worklog" directory with August.

\begin{enumerate}
\item cd ~/worklog
\item git commit -a
\item git push
\end{enumerate}

\subsubsection{LaTeX changes}
Added augustweek.tex, a .tex layout for analyzing the week's work. Not sure how to number it, for now I'll do august18th-22nd.tex, section 1.
For September, I may replace the monthly summary august.tex with all of the weekly summaries. That will be more concise. I can't see any reason to do a big summary of the whole month, it's likely better to separate out the weeks.

\subsubsection{Read Gilchrist 2014}
Strangeish. "Researchers strongly believe that genomic sequences encode a trove of biologically important information." I thought it was a given. Is there a faction of biologists who don't believe in DNA?

A question strikes me: What do we already know about CUB, and what are we going to learn about CUB? 

``...highly expressed genes should show the strongest bias for codons with shorter pausing times and error rates [Ikemura, 1981b, 1985, Sharp and Li, 1986, 1987a]. As a result, the patterns of CUB observed within a genome should contain a significant amount of information about a geneâ€™s expression level, specifically the average rate at which proteins are synthesized from the ORF. Further, because low expression genes are under very weak selection to reduce $\eta$, their patterns of CUB should provide information on the mutational biases experienced within a genome."

\sout{As I understand it, we have one direction of data? We have data about gene expression levels, and want to produce information about mutation bias and pausing times.}
We have BOTH directions of information. 

``Using the Saccharomyces cerevisiae S288c genome as an example, we demonstrate that our model can be used to accurately estimate differences in codon specific mutation biases and contributions to $\eta$ without the need for gene expression data."


But do we also have information about pausing times, and we want to find out their expression levels?
What would be the model where we have no information about gene expression levels? Is it just trying to work backwards from the results of the first model?

Answer: We have both, and can use the program to solve for either of them. With or without $\vec{X}$ (the $\phi$ values), there are (apparently) reliable results for  mutation bias and ROC. 

Questions:
\begin{itemize}
\item What is $\vec{X}$? Based on the reading, $\vec{X}$ is a set of estimates for $\phi$, but there is a sizable section about the estimation of $\phi$ given $\vec{X}$. Why isn't that estimation perfect? what am I missing here?
\item What is the input to the code when given no $\vec{X}$ estimates? Does the code first approximate the protein synthesis rate (ala Figure 3(a))? Does it use information about the ROC or NSE?
\item $\phi$ is more closely correlated with and without $vec{X}$ for higher values of $\phi$. That seems promising? When actually applied to the genome, (say S.cerevisiae), the model will mainly be dealing with genes with higher expression levels (it's just math). Is this valid? Also, why does $\delta\eta$ seem to be LESS accurate at higher expression levels? It's not unsurprising, but it is a contrast.
\item Does any of the documentation have things about the code structure? Dr. Gilchrist said it would be in the supplemental materials, but I still have no idea what B is, or drawBConditional
\end{itemize}





\subsubsection{REU Results}

\subsubsection{Workflow for R}


\subsection{Future Goals}
\begin{enumerate}
\item Continue analyzing my.cubappr.r
\item Look for Wei-Chen code on the NSE model
\item Look at REU results (esp. a1 discussion?)
\item Look into workflow programs for R
\item Read about Data Structures in R user manual
\end{enumerate}


\end{document} %End of day document, REMOVE
